### ISSUE TITLE
[A14] Low Screen Brightness after Userspace Reboot

### ISSUE THREAD

**Issue by bluetree2009** [opa334/Dopamine#123]
I have also removed the jailbreak environment and tried re-jailbreaking.

**Comment by bluetree2009** [comment]
I heard from some netizens that the brightness was adjusted to the lowest level, making it unable to automatically adjust to a visible level.

**Comment by purp0s3** [comment]
> I heard from some netizens that the brightness was adjusted to the lowest level, making it unable to automatically adjust to a visible level.

I'm on the same version and device as you.

It isn't the brightness that causes it, I was able to unlock my phone via FaceID, open cc, long press and adjust brightness and volume and turn off the device (I know it all worked because of haptic feedback).

The Fugu beta releases had a userspace reboot button instead of doing it automatically so what I did was always just respring via TrollStore afterwards.

Since Dopamine release it has been 30/70 success rate, most of the time the screen goes black, I have to turn off and do it all again.

I can also confirm it isn't a tweak issue (eve and opa said this a couple of times) as I have tested with a clean environment and got the same black screen.

**Comment by opa334** [comment]
This must be stock bug. But anyhow I will look into this in more detail with a tester soon.

**Comment by purp0s3** [comment]
> This must be stock bug. But anyhow I will look into this in more detail with a tester soon.

An update regarding this issue, ever since Dopamine 1.0.4 and ElleKit 0.6.1, userspace reboots works as they should. Same device, version and tweaks.

**Comment by yangeya** [comment]
This bug still exists    ios15.4.1 iphone12 Dopamine 1.0.5

**Comment by Vitamoon** [comment]
Same for me, 15.4.1 iPhone 12 (13,2) on Dopamine 1.0.5; this screen problem has occurred on every build of Dopamine I have used (all of them from 1.0.0).

**Comment by ajhawley** [comment]
iOS 15.4.1 on iPhone 12 and I have the same issue with the latest dopamine v1.1

have to keep restarting device and retrying, takes about 4 times to get it to work

**Comment by bluetree2009** [comment]
> retrying

yes ，the same issue~

**Comment by Vorshim92** [comment]
me too! I'm on iphone 12 ios 15.4. 9/10 i got blackscreen after jailbreak with touchscreen working (just press where there is the torch and see if it's working). I already try to remove jailbreak enviroment many times but everytime I install ellekit after a new installation I got this problem

**Comment by nice2cu1** [comment]
Same version and device,Dopamine v1.1.2.This bug still exists. So, when will this bug be fix? I can't use my phone at all after the jailbreak is completed.

**Comment by ajhawley** [comment]
I’ve not had this happen to me for a few weeks now, it’s obviously caused by something else and not dopamine, currently on v1.1.2 and not had this issue

**Comment by opa334** [comment]
> Same version and device,Dopamine v1.1.2.This bug still exists. So, when will this bug be fix? I can't use my phone at all after the jailbreak is completed.

the fix is to just respring manually after the userspace reboot

**Comment by Vorshim92** [comment]
Yes! Just make a shortcut for siri and when you are in the black lockscreen just press the power button for siri and send the command for the respring ![image](https://github.com/opa334/Dopamine/assets/7747380/ddb114fa-913d-4068-8f6e-29ebf69fb04b)

**Comment by bluetree2009** [comment]
how  let siri do the respring?




Vorshim92 ***@***.***> 于2023年7月20日周四 22:21写道：

**Comment by nice2cu1** [comment]
update:On my phone, killall SpringBoard doesn't work, the screen remains black after running the command. But I found another way, if you have a Mac, you can connect the USB cable to the Mac, use QuickTime Player to set the display to your iPhone, then press the power button to make the screen light up (I can only tell by feel iPhone screen is lit), when the phone is connected to QuickTime Player, the screen will return to its original state. NOTE: Maybe this only works on my phone. The translation is from google, so sorry for confusing you

**Comment by Vorshim92** [comment]
> update:On my phone, killall SpringBoard doesn't work, the screen remains black after running the command. But I found another way, if you have a Mac, you can connect the USB cable to the Mac, use QuickTime Player to set the display to your iPhone, then press the power button to make the screen light up (I can only tell by feel iPhone screen is lit), when the phone is connected to QuickTime Player, the screen will return to its original state. NOTE: Maybe this only works on my phone. The translation is from google, so sorry for confusing you

Hi guys! I found on discord a 100% solution working! 


I've tried everything too, attempting to send the simple respring command via terminal through SSH on PC or with Siri as shown in the picture 'killall Springboard,' but nothing. However, it's evident that the respring executed by this app works on a different level, and I assure you it ALWAYS works. 100%. Create a shortcut for Siri that launches the app I've attached below (which you obviously need to install through Trollstore), and you're done. When you find yourself on the lock screen with a completely black screen, press the power button for a few seconds to activate Siri and give the command for your shortcut (mine, for example, is 'Launch Instarespring')


[Instaspring app link](https://github.com/haxi0/InstaSpring/releases/download/1.0/InstaSpring.1.0.Fix.ipa)

![image](https://github.com/opa334/Dopamine/assets/7747380/8aff121f-914c-4438-816d-d7cf146c72a3)

**Comment by invalidunit** [comment]
I wrote a simple daemon that should slightly alleviate this issue by respring the phone once after rebooting the user space.
No action is needed after rebooting the user space, and it should finish automatically.
Install "respring after reboot userspace" from 
https://invalidunit.github.io/repo/

**Comment by jalsco** [comment]
Yes I get the "screen darkness" problem also (iOS 15.4.1 on 12 Pro Max) . A simple "restart Springboard" seems to fix it.  Can barely make out the text, but knowing it is 2nd menu item from top on the Dopamine screen is enough to be able to do a Respring.

**Comment by OnesuchDev** [comment]
A potential way to work around this within Dopamine without finding the root cause would be for code injected into launchd to check whether SpringBoard is being launched for the first time, and if so, to inject a tiny tweak into it causing it to exit(0) when it has finished launching.

**Comment by opa334** [comment]
As far as I know this is no longer an issue for some reason in Dopamine 2.x, so closing (feel free to tell me if I'm wrong)

**Comment by OnesuchDev** [comment]
Good to know. I don't use Dopamine myself, but wanted to suggest this idea
anyway.

**Comment by authorisation** [comment]
This still happens to me everytime I update dopamine or do a Userspace reboot. A springboard restart from the Dopamine app fixes it. I am on A14 15.5

**Comment by 471309682** [comment]
> 据我所知，由于某种原因，这在 Dopamine 2.x 中不再是问题，所以结束（如果我错了，请随时告诉我）

iOS 15.4.1 iPhone 12 dopamine 2.2 happened again.

**Comment by hostedbyjustus** [comment]
Does anyone know if this is a fixed issue? When I was maining my 12 mini this was extremely annoying.

**Comment by invalidunit** [comment]
> Does anyone know if this is a fixed issue? When I was maining my 12 mini this was extremely annoying.

try this
https://github.com/opa334/Dopamine/issues/123#issuecomment-1790147781

**Comment by hostedbyjustus** [comment]
In theory that's just a bypass instead of a proper solution

**Comment by hostedbyjustus** [comment]
@bluetree2009 Are you still experiencing this issue? I'm not using my 12 mini that often but the last time I checked everything was fine. If it's the same for you, this big issue could finally be closed.

EDIT: Unfortunately I ran into this issue again making this a still ongoing issue.
